[docs] Add auto-generating permission tables for android and ios

byCedric opened the pull request:
# Why

We need to improve the documentation to include permissions for (at least) three types of developers:

1. **Managed users** - These developers only use Expo Go and might not have interacted with the native configuration at all.
2. **Bare users** - These developers should have a better understanding of the permissions and how to configure them natively.
3. **Unimodules users** - Overlaps with bare, but without Expo tooling (just the library). They should be very familiar with the permissions.

# How

@brentvatne and I intend to add these tables to our API pages using a structure like this. (Wording might change though).

<details>
<summary>Big image, click to expand</summary>
<img width="672" alt="Screenshot 2021-07-21 at 22 56 12" src="https://user-images.githubusercontent.com/1203991/126558970-ea34ae61-a409-4b95-83b1-64eb88741f36.png">
</details>

This PR adds the auto-generating tables, which [use data scraped from these scripts](https://gist.github.com/byCedric/8cc3372608962708a022b5dcc6231569). Will add this to the docs in a separate PR.

# Usage

<details>
  <summary>Just adding the components with the permission keys it should render is enough.</summary>
  <img width="675" alt="Screenshot 2021-07-21 at 22 59 30" src="https://user-images.githubusercontent.com/1203991/126559195-129872b7-347a-443a-bca1-7b20280385cb.png" />
</details>

# Examples

<details>
  <summary>Android location permissions table</summary>
  <img width="675" alt="Screenshot 2021-07-21 at 22 59 30" src="https://user-images.githubusercontent.com/1203991/126559994-4e1fed37-63e2-4e87-9e48-5e15545b138b.png" />
</details>

<details>
  <summary>Android permissions table with a deprecated permission</summary>
  <img width="672" alt="Screenshot 2021-07-21 at 22 59 15" src="https://user-images.githubusercontent.com/1203991/126560105-f633d899-c8e9-4673-beb6-b029fa71de0b.png">

</details>

<details>
  <summary>Simple iOS usage descriptions</summary>
  <img width="675" alt="Screenshot 2021-07-21 at 22 59 30" src="https://user-images.githubusercontent.com/1203991/126559476-5c547d7f-3a60-4b6c-84ae-d88041d9ae33.png" />
</details>

# Other stuff

- Might be nice to add a `source: <page we scraped>` to direct users for more info
- We have to add a link per permission, but the docs don't have the right icons yet
- We don't have a tooltip atm, added a title for "more info" about the granted by table property

# Test Plan

Add it to a docs page and check it out.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

byCedric committed:
[docs] Add auto-generating permission tables for android and ios

byCedric committed:
[docs] Add proper question mark icon and tweaked android descriptions

Simek commented:
Great improvement and nice implementation. 👍 

I really like the design idea behind deprecated permissions, might borrow it for the deprecated content generated in the API docs. 😉

byCedric commented:
Feedback from Brent:
- Add link anchors to the permissions
- Add optional additions/overwrites to the tables

byCedric committed:
[docs] Hide the granted by column on Android permissions

This would still be a useful feature to add, but it has to be communicated properly

byCedric committed:
[docs] Allow permission customization in the permissions table

byCedric committed:
[docs] Add anchor links to the permissions

byCedric committed:
[docs] Add note about commented code in Android permissions table

byCedric committed:
[docs] Rename descriptionLong to explanation and add documentation

byCedric commented:
- Anchor links are added and kept out of the table of contents.
- You can overwrite each property by using an object instead of the permission name.

Overwriting is pretty simple, you can supply an object instead of just the name string. This object must follow the type definition from the `~/components/plugins/permissions/data/index.ts` (both `AndroidPermission` and `IOSPermission`). 

> When you use an object, the `name` has to match the permission you want to use. You can use all other properties to overwrite the data from the json files.

```
// This will use all data from the permissions json files
<AndroidPermissions permissions={['CAMERA']} />

// This will overwrite the (short) description
<AndroidPermissions permissions={[{ name: 'CAMERA', description: 'Something context related'}]} />

// This will remove any warning
<AndroidPermissions permissions={[{ name: 'CAMERA', warning: null }]} />
```
